cdisc_rules_engine/services/factory_interface.py

```diff
@@ -0,0 +1,13 @@
+from abc import ABC, abstractmethod
+
+
+class FactoryInterface(ABC):
+    @classmethod
+    @abstractmethod
+    def register_service(cls, name: str, service) -> None:
+        pass
+
+    @classmethod
```

Comment:
This method should be an instance method because a data service might need a config and cache which are stored in factory's self attributes